src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
+
+    if (action.type === 'addProperties') {
+      for (const key in action.extraData) {
+        newState[key] = action.extraData[key];
+      }
+    }
```

Comment:
The action of copying the currentState into newState is repeated in every case of the action type. To adhere to the DRY principle, consider moving this action outside of the switch-case structure and perform it once after determining the action to take.